pkg/experiment/compactor/compaction_worker.go

```diff
@@ -292,6 +302,76 @@ func (w *Worker) startJob(ctx context.Context, job *metastorev1.CompactionJob) *
 	return job.Status
 }
 
+func (w *Worker) exportMetricsFromCompactedBlocks(ctx context.Context, compacted []*metastorev1.BlockMeta) {
+	functions := map[string]*queryv1.FunctionList{
+		// TODO:
+		// This must be richer. First, it should be split by tenant.
+		// Also, we could have functions associated to service_name
+		// while others are just collected generally no matter what
+		// service_name we handle
+		"pyroscope": {
+			Functions: []string{
+				"net/http.HandlerFunc.ServeHTTP",
+				"runtime.gcBgMarkWorker",
+			},
+		},
+		"ride-sharing-app": {
+			Functions: []string{
+				"net/http.HandlerFunc.ServeHTTP",
+				"runtime.gcBgMarkWorker",
+			},
+		},
+	}
```

Comment:
Metrics to export need to be from config. 

This PoC export every dimension/profile-type of every tenant and service name, and the functions configured in this map. This should all be configurable.